### PR TITLE
new(tests): EOF validation CALLF stack overflow

### PR DESCRIPTION
**Pull request by hugo-dc**
## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened: https://github.com/ethereum/tests/pull/1474
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by hugo-dc** [comment]
I have moved the tests into the file containing the existing callf tests as you suggested.

**Review by pdobacz — APPROVED** [review]

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py [hunk 1]
```diff
@@ -669,6 +669,310 @@ def test_callf_stack_height_limit_exceeded(eof_test, callee_outputs):
...
+    container = Container(
+        sections=[
+            Section.Code(
+                code=Op.PUSH0
+                + Op.PUSH1[0]
+                + Op.RJUMPI[2]
+                + Op.PUSH0 * 2
+                + Op.PUSH1[1] * 509
+                + Op.CALLF[1]
+                + Op.STOP,
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py:729]:
```suggestion
                code=Op.RJUMPI[2](0)
                + Op.PUSH0 * (MAX_RUNTIME_OPERAND_STACK_HEIGHT // 2)
                + Op.CALLF[1]
                + Op.STOP,
```
I feel like this makes it a bit more clear what's in the stack before making the call.

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py [hunk 2]
```diff
@@ -669,6 +669,310 @@ def test_callf_stack_height_limit_exceeded(eof_test, callee_outputs):
...
+        name="callf_stack_overflow_variable_stack_3",
+        sections=[
+            Section.Code(
+                code=Op.PUSH0
+                + Op.PUSH1[0]
+                + Op.RJUMPI[2]
+                + Op.PUSH0 * 2
+                + Op.PUSH1[1] * 1020
+                + Op.CALLF[1]
+                + Op.STOP,
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py:759]:
```suggestion
                code=Op.RJUMPI[2](0)
                + Op.PUSH0 * (MAX_RUNTIME_OPERAND_STACK_HEIGHT - 1)
                + Op.CALLF[1]
                + Op.STOP,
```
Similar to previous comment.

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py (hunk 1 above)
Comment (marioevz) [reply tests/osaka/eip7692_eof_v1/eip4750_functions/test_code_validation.py:729]: Also I think we should create a variant of this test where the overflow happens on the jumping branch.

Something along the lines of:
```python
Op.PUSH0 * 2
+ Op.RJUMPI[2](0)
+ Op.POP * 2
+ Op.PUSH0 * (MAX_RUNTIME_OPERAND_STACK_HEIGHT // 2)
+ Op.CALLF[1]
+ Op.STOP
```

**Review by marioevz — COMMENTED** [review]
Thanks for implementing these!

I left a couple of comments and a suggestion for an extra case that I think should give us extra coverage.

**Review by marioevz — APPROVED** [review]
LGTM, thanks for the changes!
